src/v2/providers/storage.ts

```diff
@@ -0,0 +1,389 @@
+   */
+  id?: string;
+  /**
+   * The kind of item this is. For objects, this is always "storage#object".
+   */
+  kind?: string;
+  /**
+   * MD5 hash of the data; encoded using base64 as per
+   * [https://tools.ietf.org/html/rfc4648#section-4][RFC 4648 §4]. For more
+   * information about using the MD5 hash, see
+   * [https://cloud.google.com/storage/docs/hashes-etags#_JSONAPI][Hashes and
+   * ETags: Best Practices].
+   */
+  md5Hash?: string;
+  /**
+   * Media download link.
+   */
+  mediaLink?: string;
+  /**
+   * User-provided metadata, in key/value pairs.
+   */
+  metadata?: { [key: string]: string };
+  /**
+   * The version of the metadata for this object at this generation. Used for
+   * preconditions and for detecting changes in metadata. A metageneration
+   * number is only meaningful in the context of a particular generation of a
+   * particular object.
+   */
+  metageneration?: number;
+  /**
+   * The name of the object.
+   */
+  name?: string;
+  /**
+   * The link to this object.
+   */
+  selfLink?: string;
+  /**
+   * Content-Length of the object data in bytes, matching
+   * [https://tools.ietf.org/html/rfc7230#section-3.3.2][RFC 7230 §3.3.2].
+   */
+  size?: number;
+  /**
+   * Storage class of the object.
+   */
+  storageClass?: string;
+  /**
+   * The creation time of the object.
+   * Attempting to set this field will result in an error.
+   */
+  timeCreated?: Date | string;
+  /**
+   * The deletion time of the object. Will be returned if and only if this
+   * version of the object has been deleted.
+   */
+  timeDeleted?: Date | string;
+  /**
+   * The time at which the object's storage class was last changed.
+   */
+  timeStorageClassUpdated?: Date | string;
+  /**
+   * The modification time of the object metadata.
+   */
+  updated?: Date | string;
+}
+
+/**
+ * Metadata of customer-supplied encryption key, if the object is encrypted by
+ * such a key.
+ */
+export interface CustomerEncryption {
+  /**
+   * The encryption algorithm.
+   */
+  encryptionAlgorithm?: string;
+  /**
+   * SHA256 hash value of the encryption key.
+   */
+  keySha256?: string;
+}
+
+/** @hidden */
+export const service = 'storage.googleapis.com';
+/** @hidden */
+export const archivedEvent = 'google.cloud.storage.object.v1.archived';
+/** @hidden */
+export const finalizedEvent = 'google.cloud.storage.object.v1.finalized';
+/** @hidden */
+export const deletedEvent = 'google.cloud.storage.object.v1.deleted';
+/** @hidden */
+export const metadataUpdatedEvent =
+  'google.cloud.storage.object.v1.metadataUpdated';
+
+/** StorageOptions extend EventHandlerOptions with a bucket name  */
+export interface StorageOptions extends options.EventHandlerOptions {
+  bucket?: string;
+}
+
+/** Handle a storage object archived */
+export function onObjectArchived(
+  handler: (event: CloudEvent<StorageObjectData>) => any | Promise<any>
+): CloudFunction<StorageObjectData>;
+
+/** Handle a storage object archived */
```

Comment:
ugh does ESLint for us to repeat JSDoc comment on each of the overloaded function 🤦‍♂️ ?

Reply:
I was following what we did with pubsub, but since GCS has a lot more overloads, I removed the repeated ones and left the top comment.